pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -309,21 +308,21 @@ func (s *Scheduler) calculateExistingMachines(stateNodes []*state.StateNode, dae
 	})
 }
 
-func getDaemonOverhead(nodeTemplates []*MachineTemplate, daemonSetPods []*v1.Pod) map[*MachineTemplate]v1.ResourceList {
-	overhead := map[*MachineTemplate]v1.ResourceList{}
+func getDaemonOverhead(nodeClaimTemplates []*NodeClaimTemplate, daemonSetPods []*v1.Pod) map[*NodeClaimTemplate]v1.ResourceList {
+	overhead := map[*NodeClaimTemplate]v1.ResourceList{}
 
-	for _, nodeTemplate := range nodeTemplates {
+	for _, nodeClaimTemplate := range nodeClaimTemplates {
 		var daemons []*v1.Pod
 		for _, p := range daemonSetPods {
-			if err := nodeTemplate.Taints.Tolerates(p); err != nil {
+			if err := scheduling.Taints(nodeClaimTemplate.Spec.Taints).Tolerates(p); err != nil {
```

Comment:
Do we need to make a `scheduling.Taints()` call here? Isn't the nodeClaimTemplate.Spec.Taints already supposed to be this type?

Reply:
Yes, the `Tolerates` call only works with a `scheduling.Taint`. Prior to this change, we were storing taints in this state as a separate object inside of the `ProvisionerNodeTemplate`. Now, we are using the `NodeClaimTemplate` (wrapped) to store all the details, so taints are stored as `v1.Taint` which doesn't support this function and requires a cast.